[
   {
      "_links" : {
         "html" : {
            "href" : "https://github.com/bitcoin/bitcoin/pull/21907#discussion_r630076648"
         },
         "pull_request" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/21907"
         },
         "self" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/630076648"
         }
      },
      "author_association" : "MEMBER",
      "body" : "Is this specific to windows? Or could this happen with directory \"access denied\" errors on other OSes as well?",
      "commit_id" : "bd13ec6d834d4c5f20a5ff9ab41e2e9e86f32d93",
      "created_at" : "2021-05-11T11:20:59Z",
      "diff_hunk" : "@@ -19,6 +23,12 @@ std::vector<fs::path> ListDatabases(const fs::path& wallet_dir)\n     for (auto it = fs::recursive_directory_iterator(wallet_dir, ec); it != fs::recursive_directory_iterator(); it.increment(ec)) {\n         if (ec) {\n             LogPrintf(\"%s: %s %s\\n\", __func__, ec.message(), it->path().string());\n+#ifdef WIN32\n+            if (ec.value() == boost::system::windows_error::access_denied) {",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/21907#discussion_r630076648",
      "id" : 630076648,
      "line" : 27,
      "node_id" : "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDYzMDA3NjY0OA==",
      "original_commit_id" : "bd13ec6d834d4c5f20a5ff9ab41e2e9e86f32d93",
      "original_line" : 27,
      "original_position" : 16,
      "original_start_line" : null,
      "path" : "src/wallet/db.cpp",
      "position" : 16,
      "pull_request_review_id" : 656584462,
      "pull_request_url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/21907",
      "side" : "RIGHT",
      "start_line" : null,
      "start_side" : null,
      "updated_at" : "2021-05-11T11:20:59Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/630076648",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/126646?v=4",
         "events_url" : "https://api.github.com/users/laanwj/events{/privacy}",
         "followers_url" : "https://api.github.com/users/laanwj/followers",
         "following_url" : "https://api.github.com/users/laanwj/following{/other_user}",
         "gists_url" : "https://api.github.com/users/laanwj/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/laanwj",
         "id" : 126646,
         "login" : "laanwj",
         "node_id" : "MDQ6VXNlcjEyNjY0Ng==",
         "organizations_url" : "https://api.github.com/users/laanwj/orgs",
         "received_events_url" : "https://api.github.com/users/laanwj/received_events",
         "repos_url" : "https://api.github.com/users/laanwj/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/laanwj/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/laanwj/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/laanwj"
      }
   },
   {
      "_links" : {
         "html" : {
            "href" : "https://github.com/bitcoin/bitcoin/pull/21907#discussion_r630096967"
         },
         "pull_request" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/21907"
         },
         "self" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/630096967"
         }
      },
      "author_association" : "MEMBER",
      "body" : "> Is this specific to windows?\r\n\r\nThe error code is specific to windows.\r\n\r\n> Or could this happen with directory \"access denied\" errors on other OSes as well?\r\n\r\nYes, it could. But on other OSes the reason of \"access denied\" errors is a bad setup that a user can/should fix. On windows this situation seems common and not always fixable by a user.\r\n\r\n",
      "commit_id" : "bd13ec6d834d4c5f20a5ff9ab41e2e9e86f32d93",
      "created_at" : "2021-05-11T11:53:53Z",
      "diff_hunk" : "@@ -19,6 +23,12 @@ std::vector<fs::path> ListDatabases(const fs::path& wallet_dir)\n     for (auto it = fs::recursive_directory_iterator(wallet_dir, ec); it != fs::recursive_directory_iterator(); it.increment(ec)) {\n         if (ec) {\n             LogPrintf(\"%s: %s %s\\n\", __func__, ec.message(), it->path().string());\n+#ifdef WIN32\n+            if (ec.value() == boost::system::windows_error::access_denied) {",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/21907#discussion_r630096967",
      "id" : 630096967,
      "in_reply_to_id" : 630076648,
      "line" : 27,
      "node_id" : "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDYzMDA5Njk2Nw==",
      "original_commit_id" : "bd13ec6d834d4c5f20a5ff9ab41e2e9e86f32d93",
      "original_line" : 27,
      "original_position" : 16,
      "original_start_line" : null,
      "path" : "src/wallet/db.cpp",
      "position" : 16,
      "pull_request_review_id" : 656611268,
      "pull_request_url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/21907",
      "side" : "RIGHT",
      "start_line" : null,
      "start_side" : null,
      "updated_at" : "2021-05-11T11:53:53Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/630096967",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/32963518?v=4",
         "events_url" : "https://api.github.com/users/hebasto/events{/privacy}",
         "followers_url" : "https://api.github.com/users/hebasto/followers",
         "following_url" : "https://api.github.com/users/hebasto/following{/other_user}",
         "gists_url" : "https://api.github.com/users/hebasto/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/hebasto",
         "id" : 32963518,
         "login" : "hebasto",
         "node_id" : "MDQ6VXNlcjMyOTYzNTE4",
         "organizations_url" : "https://api.github.com/users/hebasto/orgs",
         "received_events_url" : "https://api.github.com/users/hebasto/received_events",
         "repos_url" : "https://api.github.com/users/hebasto/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/hebasto/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/hebasto/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/hebasto"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "Started with:\r\n\r\n`bitcoin-qt.exe -walletdir=c:\\`\r\n\r\nIs this error message expected? The app continues as normal once `OK` is clicked.\r\n\r\n![bitcoin-qt_walletdir](https://user-images.githubusercontent.com/197660/117813712-4f788a80-b25b-11eb-9e1d-976f73fa7b5a.png)\r\n\r\nOn `master` the message is a warning:\r\n\r\n![bitcoin_qt_warning](https://user-images.githubusercontent.com/197660/117814047-bac25c80-b25b-11eb-95a8-5b5aa967d1f7.png)\r\n",
      "created_at" : "2021-05-11T12:24:13Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/21907#issuecomment-838383740",
      "id" : 838383740,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/21907",
      "node_id" : "MDEyOklzc3VlQ29tbWVudDgzODM4Mzc0MA==",
      "performed_via_github_app" : null,
      "updated_at" : "2021-05-11T12:24:13Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/838383740",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/197660?v=4",
         "events_url" : "https://api.github.com/users/sipsorcery/events{/privacy}",
         "followers_url" : "https://api.github.com/users/sipsorcery/followers",
         "following_url" : "https://api.github.com/users/sipsorcery/following{/other_user}",
         "gists_url" : "https://api.github.com/users/sipsorcery/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/sipsorcery",
         "id" : 197660,
         "login" : "sipsorcery",
         "node_id" : "MDQ6VXNlcjE5NzY2MA==",
         "organizations_url" : "https://api.github.com/users/sipsorcery/orgs",
         "received_events_url" : "https://api.github.com/users/sipsorcery/received_events",
         "repos_url" : "https://api.github.com/users/sipsorcery/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/sipsorcery/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/sipsorcery/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/sipsorcery"
      }
   },
   {
      "author_association" : "MEMBER",
      "body" : "@sipsorcery \r\n\r\n> Started with:\r\n> \r\n> `bitcoin-qt.exe -walletdir=c:\\`\r\n> \r\n> Is this error message expected? The app continues as normal once `OK` is clicked.\r\n\r\nI cannot reproduce neither error message with this PR nor warning message on master. Could you share other relevant non-private options from the `bitcoin.conf`, or the `debug.log` from the beginning (where options are logged)?\r\n\r\nIn any case, `-walletdir=c:\\` on windows is a bad practice :)",
      "created_at" : "2021-05-11T16:09:46Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/21907#issuecomment-838756303",
      "id" : 838756303,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/21907",
      "node_id" : "MDEyOklzc3VlQ29tbWVudDgzODc1NjMwMw==",
      "performed_via_github_app" : null,
      "updated_at" : "2021-05-11T16:09:46Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/838756303",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/32963518?v=4",
         "events_url" : "https://api.github.com/users/hebasto/events{/privacy}",
         "followers_url" : "https://api.github.com/users/hebasto/followers",
         "following_url" : "https://api.github.com/users/hebasto/following{/other_user}",
         "gists_url" : "https://api.github.com/users/hebasto/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/hebasto",
         "id" : 32963518,
         "login" : "hebasto",
         "node_id" : "MDQ6VXNlcjMyOTYzNTE4",
         "organizations_url" : "https://api.github.com/users/hebasto/orgs",
         "received_events_url" : "https://api.github.com/users/hebasto/received_events",
         "repos_url" : "https://api.github.com/users/hebasto/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/hebasto/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/hebasto/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/hebasto"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "> I cannot reproduce neither error message with this PR nor warning message on master. Could you share other relevant non-private options from the bitcoin.conf, or the debug.log from the beginning (where options are logged)?\r\n\r\nI checked my `bitcoin.conf` at the time and made sure it was empty. I've attempted to repeat the steps with a fresh configuration and have managed to delete my whole bitcoin directory. Using the same command line I now don't get the error or warning prompt with either `bitcoin-qt` version. I'll play around a bit more to try and find what generates the previous messages.\r\n\r\n> In any case, -walletdir=c:\\ on windows is a bad practice :)\r\n\r\nSure I was only attempting to test the fix in this PR by getting the wallet logic to scan the root drive. I don't have a `D:\\` drive. \r\n\r\nWhen using `bitcoin-qt` from `master` and `bitcoin-qt.exe -regtest -walletdir=c:\\` I don't get an error about attempting to access `c:\\System Volume Information`. I'll retry with the symbolic link mentioned in #20081 although to be honest symbolic linking a root drive from within the Bitcoin app data directory seems like asking for trouble.",
      "created_at" : "2021-05-11T16:25:03Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/21907#issuecomment-838782747",
      "id" : 838782747,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/21907",
      "node_id" : "MDEyOklzc3VlQ29tbWVudDgzODc4Mjc0Nw==",
      "performed_via_github_app" : null,
      "updated_at" : "2021-05-11T16:25:03Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/838782747",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/197660?v=4",
         "events_url" : "https://api.github.com/users/sipsorcery/events{/privacy}",
         "followers_url" : "https://api.github.com/users/sipsorcery/followers",
         "following_url" : "https://api.github.com/users/sipsorcery/following{/other_user}",
         "gists_url" : "https://api.github.com/users/sipsorcery/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/sipsorcery",
         "id" : 197660,
         "login" : "sipsorcery",
         "node_id" : "MDQ6VXNlcjE5NzY2MA==",
         "organizations_url" : "https://api.github.com/users/sipsorcery/orgs",
         "received_events_url" : "https://api.github.com/users/sipsorcery/received_events",
         "repos_url" : "https://api.github.com/users/sipsorcery/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/sipsorcery/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/sipsorcery/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/sipsorcery"
      }
   },
   {
      "author_association" : "MEMBER",
      "body" : "> > In any case, -walletdir=c:\\ on windows is a bad practice :)\r\n> \r\n> Sure I was only attempting to test the fix in this PR by getting the wallet logic to scan the root drive. I don't have a `D:\\` drive.\r\n\r\nWith this PR it is possible to scan C:\\ without hanging the client, but it takes _really_ long time :)",
      "created_at" : "2021-05-11T16:49:15Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/21907#issuecomment-838816484",
      "id" : 838816484,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/21907",
      "node_id" : "MDEyOklzc3VlQ29tbWVudDgzODgxNjQ4NA==",
      "performed_via_github_app" : null,
      "updated_at" : "2021-05-11T16:49:15Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/838816484",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/32963518?v=4",
         "events_url" : "https://api.github.com/users/hebasto/events{/privacy}",
         "followers_url" : "https://api.github.com/users/hebasto/followers",
         "following_url" : "https://api.github.com/users/hebasto/following{/other_user}",
         "gists_url" : "https://api.github.com/users/hebasto/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/hebasto",
         "id" : 32963518,
         "login" : "hebasto",
         "node_id" : "MDQ6VXNlcjMyOTYzNTE4",
         "organizations_url" : "https://api.github.com/users/hebasto/orgs",
         "received_events_url" : "https://api.github.com/users/hebasto/received_events",
         "repos_url" : "https://api.github.com/users/hebasto/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/hebasto/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/hebasto/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/hebasto"
      }
   },
   {
      "author_association" : "MEMBER",
      "body" : "> I don't have a `D:\\` drive.\r\n\r\nA USB flash-drive could work, no?",
      "created_at" : "2021-05-11T16:50:56Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/21907#issuecomment-838818854",
      "id" : 838818854,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/21907",
      "node_id" : "MDEyOklzc3VlQ29tbWVudDgzODgxODg1NA==",
      "performed_via_github_app" : null,
      "updated_at" : "2021-05-11T16:51:13Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/838818854",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/32963518?v=4",
         "events_url" : "https://api.github.com/users/hebasto/events{/privacy}",
         "followers_url" : "https://api.github.com/users/hebasto/followers",
         "following_url" : "https://api.github.com/users/hebasto/following{/other_user}",
         "gists_url" : "https://api.github.com/users/hebasto/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/hebasto",
         "id" : 32963518,
         "login" : "hebasto",
         "node_id" : "MDQ6VXNlcjMyOTYzNTE4",
         "organizations_url" : "https://api.github.com/users/hebasto/orgs",
         "received_events_url" : "https://api.github.com/users/hebasto/received_events",
         "repos_url" : "https://api.github.com/users/hebasto/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/hebasto/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/hebasto/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/hebasto"
      }
   }
]
